.github/workflows/Release.yml

```diff
@@ -0,0 +1,82 @@
+name: "Publish Release"
+on:
+  workflow_dispatch:
+env:
+  build_dir: ./build
+  tarballs_dir: ./build/tarballs
+jobs:
+  release:
+    name: "Publish Pre-Release"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Needed determine latest tag
+      - name: Get Latest Tag
+        id: latest_tag
+        uses: WyriHaximus/github-action-get-previous-tag@v1
+        # TODO: in future we might need to perform backports, at which point we'll need to check
+        # that the project.version has not already been tagged 
+      - name: Check for updated Project version
+        id: project_toml
+        shell: julia --color=yes --project {0}
+        env:
+          LATEST_TAG: ${{ steps.latest_tag.outputs.tag }}
+        run: |
+          using Pkg.Types
+          project = read_project("Project.toml")
+          latest_tag = parse(VersionNumber, ENV["LATEST_TAG"])
+          @info "Project version: project.version"
+          @info "Latest tag: latest_tag"
+          project.version > latest_tag || exit(1)  # error if can't make a tag
+          open(ENV["GITHUB_OUTPUT"], "a") do io
+              println(io, "version=$(project.version)")
+              println(io, "tag=v$(project.version)")
+          end
+      - uses: actions/checkout@v4
+      - name: Download artifacts
+        id: download-artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          branch: main
+          name: ray_julia_libraries
+          path: ${{ env.tarballs_dir }}
+          workflow: CI.yml
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: "1.8"
+          arch: "x64"
+      - uses: julia-actions/cache@v1
+      - name: Update Artifacts.toml
+        shell: julia --color=yes --project {0}
+        run: |
+          using Pkg
+          Pkg.instantiate()
+          include(joinpath(pwd(), "bind_artifacts.jl"))
+          bind_artifacts()
+        working-directory: ${{ env.build_dir }}
+      - name: Commit Changes
+        id: commit
+        run: |
+          # Alternatively environment variables can be used but both author/committer need to be set
+          # https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables#_committing
+          git config user.name beacon-buddy
+          git config user.email beacon-buddy@beacon.bio
+
+          msg="Update Artifacts.toml to use ${{ steps.project_toml.outputs.version }} assets"
+          git checkout --detach
```

Comment:
I missed this first time around.  is there a reason to detach rather than branching off `main`?  I think it's better to have a direct link back to the commit that the release was cut from

Reply:
I added this. The logic here is that we should be using the HEAD as checked out by the previous action. What we should do is pass in the `ref` we want to use from the `workflow_dispatch` event.

Reply:
I'd advocate we should be using:
```yaml
- uses: actions/checkout@v3
  with:
    fetch-depth: 0  # Needed determine latest tag
    ref: main
```
If we want to be explicit (note `ref` defaults to `origin/HEAD`)

Reply:
added

Reply:
I considered having an `input` for the `workflow_dispatch` but I couldn't see in what scenario we wouldn't want to tag off `main`...unless we want to tag from a specific commit on `main` in case there's been another merge in the meantime?

Reply:
> I considered having an `input` for the `workflow_dispatch` but I couldn't see in what scenario we wouldn't want to tag off `main`...unless we want to tag from a specific commit on `main` in case there's been another merge in the meantime?

The scenario would be backports. I think we can punt on this for now.

Reply:
💡